clients/client-core/src/client/cover_traffic_stream.rs

```diff
@@ -159,9 +166,21 @@ impl LoopCoverTrafficStream<OsRng> {
             self.average_cover_message_sending_delay,
         )));
 
-        while self.next().await.is_some() {
-            self.on_new_message().await;
+        let mut shutdown = self.shutdown.clone();
+        while !shutdown.is_shutdown() {
+            tokio::select! {
+                biased;
+                _ = shutdown.recv() => {
+                    log::trace!("LoopCoverTrafficStream: Received shutdown");
```

Comment:
if we received shutdown, shouldn't we break the loop?

Reply:
When `.recv()` fires then it should toggle `shutdown.is_shutdown()`